### PR TITLE
coord: Add system configurations

### PR DESCRIPTION
**Pull request by jkosh44**
This commit adds system configuration parameters and their default
values. Additionally, this commit finishes the ALTER SYSTEM
implementation which allows us to modify system configuration
parameters.

MaterializeInc/cloud#3292

### Motivation
This PR adds a known-desirable feature.

### Tips for reviewer
 * Much of this PR is inspired by the existing session variables implementation.
 * Currently the system variables (on disk persisted configuration) and the session variables (session specific configuration) are disjoint sets. I played around with combining them into a single set with different hierarchies, but ultimately decided against it due to unnecessary complexity. If you think that they should be implemented as combined sets, then let me know. To implement it you'd need to roughly do the following steps:
   * Change the `parent` of `SessionVar` to `SystemVar`.
   * Merge `SessionVars` and `SystemVars` into a single struct.
   * When calling `SET ...` update only the `SessionVar` in the single session's struct.
   * When calling `ALTER SYSTEM ...` update only the `SystemVar` in the catalog's struct.
   * Periodically re-hydrate the `SystemVar`s in every session (maybe every query a session executes).

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - N/A

### CODE REVIEW

File: test/testdrive/session.td [hunk 1]
```diff
@@ -29,6 +29,14 @@ standard_conforming_strings on                     "Causes '...' strings to trea
 statement_timeout           "10 s"                 "Sets the maximum allowed duration of INSERT...SELECT, UPDATE, and DELETE operations."
 TimeZone                    UTC                    "Sets the time zone for displaying and interpreting time stamps (PostgreSQL)."
 transaction_isolation       "strict serializable"  "Sets the current transaction's isolation level (PostgreSQL)."
+max_clusters                10                     "The maximum number of clusters in the region (Materialize)."
```
Comment (mjibson) [inline test/testdrive/session.td:32]: Bikeshed: should these be prepended with system: `system_max_clusters`?

File: test/sqllogictest/vars.slt [hunk 2]
```diff
@@ -194,3 +194,53 @@ SHOW extra_float_digits
...
+
+query T
+SHOW max_databases
+----
+1000
+
+query T
+SHOW max_schemas_per_database
+----
+1000
```
Comment (mjibson) [inline test/sqllogictest/vars.slt:246]: Can you add a ALTER SYSTEM SET test for a var name that doesn't exist?

**Review by mjibson — APPROVED** [review]

File: test/testdrive/session.td (hunk 1 above)
Comment (jkosh44) [reply test/testdrive/session.td:32]: The names were chosen to be consistent with Postgres. This is from the doc with all the variables (https://www.notion.so/materialize/Default-system-limits-f36a447644d9461f8665180fc5d7c505):
> Limits will be exposed to end users via [system configuration parameters](https://www.notion.so/materialize/Server-configuration-parameters-c9aa8050a2884f67a30b4a34aae2d1bd?d=653c75a293b84890b39d5829366c3e6e#be931d29323a4c46ac3c10cdade785f2). They are named consistently with PostgreSQL parameters that describe limits, like max_connections and max_files_per_process.

File: test/sqllogictest/vars.slt (hunk 2 above)
Comment (jkosh44) [reply test/sqllogictest/vars.slt:246]: Added

**Pull request merged**